src/common/components/fallback-boundary/fallback-boundary.tsx

```diff
@@ -25,22 +27,28 @@ export default function FallbackBoundary({
   error = {},
   suspense = {},
 }: FallbackBoundaryProps) {
-  const { fallbackUI: errorFallbackUI = <Skeleton />, onError, onReset } = error;
+  const { fallbackUI: errorFallbackUI = <Skeleton />, onError } = error;
   const { fallbackUI: suspenseFallbackUI = <Skeleton /> } = suspense;
 
   return (
-    <ErrorBoundary
-      // errorFallbackUI가 함수일 경우 resetErrorBoundary 주입
-      fallbackRender={({ resetErrorBoundary }) => {
-        if (typeof errorFallbackUI === 'function') {
-          return errorFallbackUI(resetErrorBoundary);
-        }
-        return errorFallbackUI;
-      }}
-      onError={onError}
-      onReset={onReset}
-    >
-      <Suspense fallback={suspenseFallbackUI}>{children}</Suspense>
-    </ErrorBoundary>
+    <QueryErrorResetBoundary>
+      {({ reset }) => (
+        <ErrorBoundary
+          fallbackRender={({ resetErrorBoundary }) => {
+            if (typeof errorFallbackUI === 'function') {
+              return errorFallbackUI(() => {
+                reset();
+                resetErrorBoundary();
+              });
+            }
```

Comment:
```suggestion
            if (typeof errorFallbackUI === 'function') {
              return errorFallbackUI(() => {
                resetErrorBoundary();
              });
            }
```

errorFallbackUI가 function일 때에 reset 함수를 명시적으로 실행시켜줘야 할까요??
기존처럼 resetErrorBoundary만 넘겨도 onReset 이벤트 핸들러에서 reset 함수를 실행시키기 때문에 괜찮을 것 같습니다!